spark/src/main/scala/org/apache/spark/sql/delta/DeltaColumnMapping.scala

```diff
@@ -511,10 +511,15 @@ trait DeltaColumnMappingBase extends DeltaLogging {
       throw DeltaErrors.unsupportedColumnMappingMode(columnMappingMode.name)
     }
 
+    val referenceSchemaColumnMap: Map[String, StructField] =
+      SchemaMergingUtils.explode(referenceSchema).map { case (path, field) =>
+        QuotingUtils.quoteNameParts(path).toLowerCase() -> field
```

Comment:
We should avoid depending on the default locale.

```suggestion
        QuotingUtils.quoteNameParts(path).toLowerCase(Locale.ROOT) -> field
```